operations/fetch.go

```diff
@@ -27,7 +27,7 @@ import (
 	"github.com/urfave/cli"
 )
 
-const defaultCloneDepth = 500
+const defaultCloneDepth = 1000
```

Comment:
Isn't this change in the opposite direction of what we want to do? Do we anticipate a scenario where 500 commits is not enough but 1000 is?

Reply:
Indeed, shouldn't we try with something very small and just unshallow in the failure case?

Reply:
I tested cloning the mongodb/mongo and 10gen/mms repos on a spawnhost with no `--depth` parameter, and with `--depth` set to 1, 100, and 1000. The difference between cloning with no `--depth` and with `--depth` is appreciable, but the difference between the different values passed to `--depth` was either tiny or not in the direction I was expecting (suggesting that noise from the network/filesystem is larger than the actual difference between them).

The choice of 1000 is, of course, arbitrary. I picked it because it's a large round number and wasn't obviously worse than 1 or 100 in my tests.